lib/Reader/readerir.cpp

```diff
@@ -575,6 +576,92 @@ void GenIR::callMonitorHelper(bool IsEnter) {
                  MethodSyncHandle, (IRNode *)SyncFlag);
 }
 
+void GenIR::insertIRForUnmanagedCallFrame() {
+  const struct CORINFO_EE_INFO::InlinedCallFrameInfo &CallFrameInfo =
+      JitContext->EEInfo.inlinedCallFrameInfo;
+  LLVMContext &LLVMContext = *JitContext->LLVMContext;
+  Type *Int8Ty = Type::getInt8Ty(LLVMContext);
+  Type *Int32Ty = Type::getInt32Ty(LLVMContext);
+  Type *Int8PtrTy = getUnmanagedPointerType(Int8Ty);
+  IRBuilder<>::InsertPoint SavedInsertPoint = LLVMBuilder->saveIP();
+
+  // Mark this function as requiring a frame pointer and as using GC.
+  Function->addFnAttr("no-frame-pointer-elim-non-leaf");
```

Comment:
In `GenIR::insertIRToKeepGenericContextAlive` I did this by modifying the target machine options.
```
  // This method now requires a frame pointer.
  TargetMachine *TM = JitContext->EE->getTargetMachine();
  TM->Options.NoFramePointerElim = true;
```
We should agree on which is the right mechanism and make it a helper and use it in both places.